src/components/async-button/async-button.test.tsx

```diff
@@ -0,0 +1,46 @@
+import { cleanup, fireEvent, screen } from "solid-testing-library";
+
+import { renderWithHopeProvider } from "@/utils/test-utils";
+
+import { buttonStyles } from "../button/button.styles";
+import { AsyncButton } from "./async-button";
+
+describe("AsyncButton", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.useRealTimers();
+    cleanup();
+  });
+
+  it("should render", () => {
+    // act
+    renderWithHopeProvider(() => <AsyncButton />);
+    const button = screen.getByRole("button");
+
+    // assert
+    expect(button).toBeInTheDocument();
+  });
+
+  it("should change state to loading and back when clicked", () => {
+    // setup
+    jest.useFakeTimers();
+    jest.spyOn(global, "setTimeout");
+    const buttonNotLoadingClass = buttonStyles({ loading: false });
+    const buttonLoadingClass = buttonStyles({ loading: false });
```

Comment:
I guest it should be `loading: true`

Reply:
the problem with that is buttonLoadingClass.className includes class in buttonNotLoadingClass.className and test will not show whether loading is finished or not. I changed it to having all classes of buttonLoadingClass.className or NOT having all classes. It is kind of verbose, but does the trick. Also I figured out we do not need fake timers for the component test.